src/ACEtk/block/MultiDistributionData/src/ctor.hpp

```diff
@@ -1,15 +1,29 @@
 MultiDistributionData() = default;
 
 MultiDistributionData( const MultiDistributionData& base ) :
-  Base( base ) {
+  Base( base ), locb_( base.locb_ ),
+  probabilities_( base.probabilities_ ),
+  distributions_( base.distributions_ ) {
 
-  this->generateBlocks();
+  if ( Base::owner() ) {
+
+    this->probabilities_.clear();
+    this->distributions_.clear();
+    this->generateBlocks();
+  }
 }
 
 MultiDistributionData( MultiDistributionData&& base ) :
-  Base( std::move( base ) ) {
+  Base( std::move( base ) ), locb_( base.locb_ ),
+  probabilities_( std::move( base.probabilities_ ) ),
```

Comment:
As I understand it, the MultiDistributionData move constructor casts ```base``` to an rvalue using std::move, then passes the result to the Base move constructor, which involves an implicit cast from MultiDistributionData&& to Base&&. The Base move constructor might steal the guts of the base class members of ```base```, but it can't touch the guts of the derived class members because it only sees a Base&&.

I therefore think this is safe - as long as I do not need base class members from that point on.